src/util/Location.ts

```diff
@@ -45,10 +45,31 @@ export class Location {
         return fs.ensureDir(this.basePath);
     }
 
-    public async unlinkIfExists(): Promise<void> {
-        if (await this.exists()) {
-            await fs.unlink(this.basePath);
-        }
+    /**
+     * Remove a path, retrying a few times after a delay in case of error.
+     * If the path does not exist, silently does nothing.
+     */
+    public async remove(maxRetries = 3, retryDelay = 1): Promise<void> {
```

Comment:
I would rename `retryDelay` to `retryDelaySec` (or `retryDelayMs` and remove the multiplication by 1000)